### PR TITLE
CLI dev flags

### PR DESCRIPTION
**Pull request by dherman**
Enables two new flags for `neon new`, both of which should become useful for development and experimental work:

- `neon new --neon=<version-or-path>`: specify a non-default version of Neon or path to a local Neon project directory on the filesystem
- `neon new --features=<neon-experimental-features-list>`: specify a space-separated list of experimental Neon features to enable

These should be useful for contributing or testing when we start developing N-API support, so that we can put the experimental support behind a feature flag until it stabilizes.

### CODE REVIEW

File: cli/src/ops/neon_new.ts [hunk 1]
```diff
@@ -117,13 +141,39 @@ export default async function wizard(pwd: string, name: string) {
...
+    simple: boolean,
+    paths?: { neon: string, "neon-build": string },
+    version?: string,
+    features?: Array<string>
+  } = { simple: neonVersion.type !== 'path' && !features };
+
+  if (neonVersion.type === 'path') {
+    libs.paths = {
+      neon: path.resolve(neonVersion.value),
+      "neon-build": path.resolve(neonVersion.value, "crates", "neon-build")
```
Comment (kjvalencik) [inline cli/src/ops/neon_new.ts]: What do you think about using [`path.relative`](https://nodejs.org/api/path.html#path_path_relative_from_to) here? `path.resolve` will produce an absolute path which isn't very friendly to switching across environments (e.g., docker).

File: cli/templates/package.json.hbs [hunk 2]
```diff
@@ -16,7 +16,7 @@
   "license": "{{project.license}}",
 {{/if}}
   "dependencies": {
```
Comment (kjvalencik) [inline cli/templates/package.json.hbs]: Not part of this change, but shouldn't this be in `devDependencies`? I think most neon projects would be top level applications and not libraries.

File: cli/src/ops/neon_new.ts [hunk 3]
```diff
@@ -117,13 +141,39 @@ export default async function wizard(pwd: string, name: string) {
...
+  }
+
+  if (features) {
+    libs.features = features.split(/\s+/);
+  }
+
+  let cli = neonVersion.type === 'version'
+    ? "^" + neonVersion.value
+    : neonVersion.type === 'path'
+    ? path.resolve(neonVersion.value, "cli")
```
Comment (kjvalencik) [inline cli/src/ops/neon_new.ts]: Should this be prefixed with `file:`?

File: cli/src/ops/neon_new.ts (hunk 1 above)
Comment (dherman) [reply cli/src/ops/neon_new.ts]: How about we go with: it'll generate a relative path if you pass a relative path and an absolute path if you pass an absolute path.

File: cli/templates/package.json.hbs (hunk 2 above)
Comment (dherman) [reply cli/templates/package.json.hbs]: As long as `neon build` is needed for running the build, it has to be in `"dependencies"`. If we can get rid of `neon build` (which would be awesome!) then we can move it to `"devDependencies"`.
